src/core/BondList.hpp

```diff
@@ -171,7 +171,8 @@ class BondList {
 
     return *this;
   }
-  BondList &operator=(BondList &&rhs) noexcept {
+  // NOLINTNEXTLINE(bugprone-exception-escape)
+  BondList &operator=(BondList &&rhs) {
```

Comment:
This should probably be fixed, because it has performances effects if `BondList` is nothrow-move-assignable... Do you remember what the problem here was?

Reply:
I accidentally removed the `noexcept`, good that you spotted that.

Reply:
```
/home/espresso/espresso/src/core/BondList.hpp:174:13: error: an
exception may be thrown in function 'operator=' which should not
throw exceptions [bugprone-exception-escape,-warnings-as-errors]
  BondList &operator=(BondList &&rhs) noexcept {
            ^
/home/espresso/espresso/src/core/BondList.hpp:227:15: error: an
exception may be thrown in function 'swap' which should not throw
exceptions [bugprone-exception-escape,-warnings-as-errors]
  friend void swap(BondList &lhs, BondList &rhs) {
              ^
```

Reply:
Ah, probably because the respective operations in the boost container can throw. I think since there is no performance benefit from them at the moment we could just switch the backend to `std::vector`. I'm a bit uncomfortable about all the linter exceptions...

Reply:
Me too, but I only added these `NOLINT` in places where I couldn't find a quick fix. Hopefully we can address them in the future.